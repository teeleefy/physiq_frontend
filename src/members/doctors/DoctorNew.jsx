import { Form, FormGroup, Input, Button, Label, Alert } from "reactstrap";
import { useState, useContext } from "react";
import {MemberContext} from "../../auth/UserContext.js";
import PhysiqApi from "../../Api.js";
import { useNavigate } from "react-router-dom";
import '../styles/Doctor.css'
import { toast } from 'react-toastify';

function DoctorNew(){
    let navigate = useNavigate();
    const { currentMember } = useContext(MemberContext);
    const [formMessages, setFormMessages] = useState([]);
    const [updateSuccess, setUpdateSuccess] =useState();
    const [formData, setFormData] = useState({name: "", specialty: "", clinic: "", address:"", phone:"", notes: ""});

    async function addDoctor() {
        try {
          let newDoctor= await PhysiqApi.addMemberDoctor(currentMember.id, formData);
          return { success: true };
        } catch (errors) {
          console.error("Add Doctor failed", errors);
          return { success: false, errors };
        }
      }

    async function handleSubmit(evt){
        evt.preventDefault();
        // console.log(formData);
        let result = await addDoctor();
        
        if(result.success){
            toast.success('Doctor Added!', {
                position: "top-center",
                autoClose: 2000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: true,
                draggable: true,
                progress: undefined,
                theme: "dark",
                }); 
           setFormMessages(['Doctor Added!'])
           setUpdateSuccess(true);
           navigate("..", { relative: "path"});
        }
        else{
            setFormMessages(result.errors);
            setUpdateSuccess(false);
        }
    };

  
    /** Update local state w/curr state of input elem */
  
    const handleChange = evt => {
      const { name, value }= evt.target;
      setFormData(fData => ({
        ...fData,
        [name]: value
      }));
    };

    return(
        <>
            <Form className="DoctorNew-Form m-4 pt-1" >
            <h1 className="Doctor-h1">Add Doctor:</h1>
            <p className="text-secondary">Fields marked with <span className="text-danger">*</span> are required.</p>
            <FormGroup>
                <Label className="Doctor-label" for="name" >
                    <b>Doctor Name</b><span className="text-danger">*</span>
                </Label>
                <Input
                id="name"
                name="name"
                maxLength={100}
                value={formData.name}
                placeholder="Enter doctor name..."
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
            <FormGroup>
                <Label className="Doctor-label" for="specialty">
                    <b>Specialty</b><span className="text-danger">*</span>
                </Label>
                <Input
                id="specialty"
                name="specialty"
                maxLength={100}
                value={formData.specialty}
                placeholder="Enter specialty..."
                type="text"
                onChange={handleChange}
                />
                </FormGroup>
            <FormGroup>
                <Label className="Doctor-label" for="clinic">
                    <b>Clinic</b>
                </Label>
                <Input
                id="clinic"
                name="clinic"
                maxLength={100}
                value={formData.clinic}
                placeholder="Enter clinic..."
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
            <FormGroup>
                <Label className="Doctor-label" for="address">
                    <b>Address</b>
                </Label>
                <Input
                id="address"
                name="address"
                maxLength={100}
                value={formData.address}
                placeholder="Enter address..."
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
            <FormGroup>
                <Label className="Doctor-label" for="phone">
                    <b>Phone</b>
                </Label>
                <Input
                id="phone"
                name="phone"
                maxLength={25}
                value={formData.phone}
                placeholder="Enter phone..."
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
            <FormGroup>
                <Label className="Doctor-label" for="notes">
                    <b>Notes</b>
                </Label>
                <Input
                id="notes"
                name="notes"
                maxLength={250}
                value={formData.notes}
                placeholder="Enter notes..."
                type="textarea"
                onChange={handleChange}
                />
            </FormGroup>
            <p className="text-secondary">{formData.notes.length}/250 Characters</p>
                {formMessages.length
                    ? formMessages.map(msg => <Alert color={updateSuccess ? "success": "danger"}>{msg}</Alert>)
                    : null
                }
                
                <Button className="btn-dark" onClick={handleSubmit}>Add Doctor</Button>
            </Form>
        </>
    )}
    
export default DoctorNew;