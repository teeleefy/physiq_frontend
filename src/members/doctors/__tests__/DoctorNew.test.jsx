import { renderWithRouterAndMemberContext, screen } from '../../../../__tests__/test-tools';
import DoctorNew from '../DoctorNew'


describe('DoctorNew', () => {
  it('renders the DoctorNew component', () => {
    renderWithRouterAndMemberContext(<DoctorNew />)
    
    screen.debug(); // prints out the jsx in the DoctorNew component unto the command line
  })

  it("matches snapshot", function() {
    const { asFragment } = renderWithRouterAndMemberContext(<DoctorNew />);
    expect(asFragment()).toMatchSnapshot();
  });

})
