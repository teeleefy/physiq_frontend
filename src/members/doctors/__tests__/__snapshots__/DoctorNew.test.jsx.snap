// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DoctorNew > matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="DoctorNew-Form m-4 pt-1"
  >
    <h1
      class="Doctor-h1"
    >
      Add Doctor:
    </h1>
    <p
      class="text-secondary"
    >
      Fields marked with 
      <span
        class="text-danger"
      >
        *
      </span>
       are required.
    </p>
    <div
      class="mb-3"
    >
      <label
        class="Doctor-label form-label"
        for="name"
      >
        <b>
          Doctor Name
        </b>
        <span
          class="text-danger"
        >
          *
        </span>
      </label>
      <input
        class="form-control"
        id="name"
        maxlength="100"
        name="name"
        placeholder="Enter doctor name..."
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="Doctor-label form-label"
        for="specialty"
      >
        <b>
          Specialty
        </b>
        <span
          class="text-danger"
        >
          *
        </span>
      </label>
      <input
        class="form-control"
        id="specialty"
        maxlength="100"
        name="specialty"
        placeholder="Enter specialty..."
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="Doctor-label form-label"
        for="clinic"
      >
        <b>
          Clinic
        </b>
      </label>
      <input
        class="form-control"
        id="clinic"
        maxlength="100"
        name="clinic"
        placeholder="Enter clinic..."
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="Doctor-label form-label"
        for="address"
      >
        <b>
          Address
        </b>
      </label>
      <input
        class="form-control"
        id="address"
        maxlength="100"
        name="address"
        placeholder="Enter address..."
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="Doctor-label form-label"
        for="phone"
      >
        <b>
          Phone
        </b>
      </label>
      <input
        class="form-control"
        id="phone"
        maxlength="25"
        name="phone"
        placeholder="Enter phone..."
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="Doctor-label form-label"
        for="notes"
      >
        <b>
          Notes
        </b>
      </label>
      <textarea
        class="form-control"
        id="notes"
        maxlength="250"
        name="notes"
        placeholder="Enter notes..."
      />
    </div>
    <p
      class="text-secondary"
    >
      0/250 Characters
    </p>
    <button
      class="btn-dark btn btn-secondary"
      type="button"
    >
      Add Doctor
    </button>
  </form>
</DocumentFragment>
`;
